Xamarin.Forms.Platform.UAP/Shell/ShellSectionRenderer.cs

```diff
@@ -191,7 +175,49 @@ internal void NavigateToContent(ShellNavigationSource source, ShellContent shell
 			}
 		}
 
-		NavigationTransitionInfo GetTransitionInfo(ShellNavigationSource navSource)
+		protected virtual void OnPopRequested(ShellNavigationSource source)
+		{
+			Frame.GoBack(GetTransitionInfo(source));
+		}
+
+		protected virtual void OnPopToRootRequested(ShellNavigationSource source)
+		{
+			while (Frame.BackStackDepth > 2)
+			{
+				Frame.BackStack.RemoveAt(1);
+			}
+			Frame.GoBack(GetTransitionInfo(source));
+		}
+
+		protected virtual void OnPushRequested(ShellNavigationSource source)
+		{
+			Frame.Navigate(typeof(ShellPageWrapper), GetTransitionInfo(source));
+		}
+
+		protected virtual void OnInsertRequested(ShellNavigationSource source, Page page)
+		{
+			var pageIndex = ShellSection.Stack.ToList().IndexOf(page);
+			if (pageIndex == Frame.BackStack.Count - 1)
+				Frame.Navigate(typeof(ShellPageWrapper), GetTransitionInfo(source));
+			else
+				Frame.BackStack.Insert(pageIndex, new PageStackEntry(typeof(ShellPageWrapper), null, GetTransitionInfo(source)));
```

Comment:
Do you need/want access to the Frame variable?

Reply:
for my usecase is not needed, i just need to make some actions before and after the navigation, inflating a custom transition if needed. 

If you think it could be useful to gain better control we can make it public, i'm sure it will open other possibilities even if are not needed now

Reply:
We'll just leave it for now and see what else comes up 👍